File: backend/src/main/java/com/quizmatrix/quizmatrix/model/Answer.java
package com.quizmatrix.quizmatrix.model;


import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

import jakarta.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity(name = "answer")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Answer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id_answer;
    @ManyToOne
    @JoinColumn(name = "id_question")
    private Question question;

    private String answer_text;
    Boolean isCorrect;
}

File: frontend/QuizMatrix/src/app/models/answer.ts
export class Answer {
  id_answer: number;
  id_question: number;
  answer_text: string;
  isCorrect: boolean;

  constructor(id_answer: number, id_question: number, answer_text: string, isCorrect: boolean) {
    this.id_answer = id_answer;
    this.id_question = id_question;
    this.answer_text = answer_text;
    this.isCorrect = isCorrect;
  }
}
